sql/hive/src/test/scala/org/apache/spark/sql/sources/HadoopFsRelationTest.scala

```diff
@@ -125,56 +126,58 @@ abstract class HadoopFsRelationTest extends QueryTest with SQLTestUtils with Tes
     } else {
       Seq(false)
     }
-    for (dataType <- supportedDataTypes) {
-      for (parquetDictionaryEncodingEnabled <- parquetDictionaryEncodingEnabledConfs) {
-        val extraMessage = if (isParquetDataSource) {
-          s" with parquet.enable.dictionary = $parquetDictionaryEncodingEnabled"
-        } else {
-          ""
-        }
-        logInfo(s"Testing $dataType data type$extraMessage")
-
-        val extraOptions = Map[String, String](
-          "parquet.enable.dictionary" -> parquetDictionaryEncodingEnabled.toString
-        )
-
-        withTempPath { file =>
-          val path = file.getCanonicalPath
-
-          val dataGenerator = RandomDataGenerator.forType(
-            dataType = dataType,
-            nullable = true,
-            new Random(System.nanoTime())
-          ).getOrElse {
-            fail(s"Failed to create data generator for schema $dataType")
+    withSQLConf(SQLConf.SESSION_LOCAL_TIMEZONE.key -> "UTC") {
```

Comment:
I'm a little worried here. This test is a round-trip test, do you mean if we write out a date/timestamp to json and read it back, the values will be different if session timezone is not UTC?

Reply:
It should be same, if the session local timezone doesn't change between write and read back.

Reply:
> I'm a little worried here. This test is a round-trip test ...
> It should be same, if the session local timezone doesn't change between write and read back.

Not only JSON parser/formatter involved in the loop but also converting milliseconds to Java's `Timestamp` and to something else.

Reply:
Let me look at it deeper.

Reply:
can we add some round-trip tests in `DateTimeFormatterSuite` for timestamp before year 1970?

Reply:
Not sure, this is related to 1970. I added a few round trip tests

Reply:
can you include the random seed in the test name(using `withClue`)? So that we can reproduce the bug easily and I can also help debugging.

Reply:
can we remove the timezone setting here? Then we can look at jenkens report and see which seed can reproduce the bug and debug it locally.

Reply:
> see which seed can reproduce the bug and debug it locally.

I ran it locally many times. It is almost 100% reproducible for any seed.

Reply:
What about to put the test under the flag `spark.sql.legacy.timeParser.enabled` and create a separate JIRA ticket? I would believe the bug somewhere in Spark's home made date/time functions rather than Java 8 implementation of timestamps parsing.

Reply:
SGTM. Can you create the ticket? And put a TODO here which refers to the ticket.

Reply:
Here is the ticket: https://issues.apache.org/jira/browse/SPARK-26374 and I added TODO